airbyte-webapp/src/components/connectorBuilder/manifestToBuilderForm.ts

```diff
@@ -0,0 +1,384 @@
+  if (streams === undefined || streams.length === 0) {
+    const { inputs, inferredInputOverrides } = manifestSpecAndAuthToBuilder(manifest.spec, undefined);
+    builderFormValues.inputs = inputs;
+    builderFormValues.inferredInputOverrides = inferredInputOverrides;
+
+    return builderFormValues;
+  }
+
+  assertType<SimpleRetriever>(streams[0].retriever, "SimpleRetriever", streams[0].name);
+  assertType<HttpRequester>(streams[0].retriever.requester, "HttpRequester", streams[0].name);
+  builderFormValues.global.urlBase = streams[0].retriever.requester.url_base;
+
+  const { inputs, inferredInputOverrides, auth } = manifestSpecAndAuthToBuilder(
+    manifest.spec,
+    streams[0].retriever.requester.authenticator
+  );
+  builderFormValues.inputs = inputs;
+  builderFormValues.inferredInputOverrides = inferredInputOverrides;
+  builderFormValues.global.authenticator = auth;
+
+  const serializedStreamToIndex = Object.fromEntries(streams.map((stream, index) => [JSON.stringify(stream), index]));
+  builderFormValues.streams = streams.map((stream, index) =>
+    manifestStreamToBuilder(stream, index, serializedStreamToIndex, builderFormValues.global)
+  );
+
+  return builderFormValues;
+};
+
+const manifestStreamToBuilder = (
+  stream: DeclarativeStream,
+  index: number,
+  serializedStreamToIndex: Record<string, number>,
+  builderFormGlobal: BuilderFormValues["global"]
+): BuilderStream => {
+  assertType<SimpleRetriever>(stream.retriever, "SimpleRetriever", stream.name);
+  const retriever = stream.retriever;
+
+  assertType<HttpRequester>(retriever.requester, "HttpRequester", stream.name);
+  const requester = retriever.requester;
+
+  if (!isEqual(retriever.requester.authenticator, builderFormGlobal.authenticator)) {
+    throw new ManifestCompatibilityError(stream.name, "authenticator does not match the first stream's");
+  }
+
+  if (retriever.requester.url_base !== builderFormGlobal.urlBase) {
+    throw new ManifestCompatibilityError(stream.name, "url_base does not match the first stream's");
+  }
+
+  if (retriever.name !== stream.name || requester.name !== stream.name) {
+    throw new ManifestCompatibilityError(
+      stream.name,
+      "name is not consistent across stream, retriever, and requester levels"
+    );
+  }
+
+  if (![undefined, "GET", "POST"].includes(requester.http_method)) {
+    throw new ManifestCompatibilityError(stream.name, "http_method is not GET or POST");
+  }
+
+  assertType<DpathExtractor>(retriever.record_selector.extractor, "DpathExtractor", stream.name);
+
+  if (requester.request_options_provider) {
+    assertType<InterpolatedRequestOptionsProvider>(
+      requester.request_options_provider,
+      "InterpolatedRequestOptionsProvider",
+      stream.name
+    );
+  }
+
+  return {
+    ...DEFAULT_BUILDER_STREAM_VALUES,
+    id: index.toString(),
+    name: stream.name ?? "",
+    urlPath: requester.path,
+    httpMethod: (requester.http_method as "GET" | "POST" | undefined) ?? "GET",
```

Comment:
Not broken by you, but I just realized we do not set the http_method on the requester when doing form state -> manifest conversion. This is a bug right?

Reply:
Yep, good catch. I'll include the fix in this PR since it's just one line